### PR TITLE
DL-4530 Fixed Pay SA Question Cache

### PR DESCRIPTION
**Pull request by Sir-Iyke**
DL-4530 Fixed Pay SA Question Cache

**Bug fix** 

Pay SA question was not storing the answer in cache. Fixed this so that when you use the back link and go back to the question, it retrieves the answer from the cache and pre populates the form.

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: hs_err_pid29104.log [hunk 1]
```diff
@@ -0,0 +1,513 @@
+#
```
Comment (paulanderson1234) [inline hs_err_pid29104.log:1]: What is this file - any ideas?

File: test/unit/uk/gov/hmrc/agentclientmandate/controllers/agent/PaySAQuestionControllerSpec.scala [hunk 2]
```diff
@@ -69,6 +71,19 @@ class PaySAQuestionControllerSpec extends PlaySpec  with BeforeAndAfterEach with
...
+
+      "agent requests(GET) for cached 'paySa question' view with some data saved" in new Setup {
+        viewWithAuthorisedAgentWithSomeData { result =>
+          status(result) must be(OK)
+          val document = Jsoup.parse(contentAsString(result))
+          document.title() must be("agent.paySA-question.title - GOV.UK")
+          document.getElementById("header").text() must include("agent.paySA-question.header")
+          document.getElementById("pre-header").text() must be("ated.screen-reader.section agent.add-a-client.sub-header")
+          document.getElementById("paySA_legend").text() must be("agent.paySA-question.header")
+          document.getElementById("paySA-true").attr("checked") must be("checked")
```
Comment (paulanderson1234) [inline test/unit/uk/gov/hmrc/agentclientmandate/controllers/agent/PaySAQuestionControllerSpec.scala:83]: Is there any value is checking for paySA-false as well - I know we have 100% coverage so may be overkill

File: hs_err_pid29104.log (hunk 1 above)
Comment (Sir-Iyke) [reply hs_err_pid29104.log:1]: Its a log file that gets generated when the repo has issues compiling. Someone must have commited it ages ago by accident so I deleted it.

File: test/unit/uk/gov/hmrc/agentclientmandate/controllers/agent/PaySAQuestionControllerSpec.scala (hunk 2 above)
Comment (Sir-Iyke) [reply test/unit/uk/gov/hmrc/agentclientmandate/controllers/agent/PaySAQuestionControllerSpec.scala:83]: I think you're right about it being overkill. I checked how the cache was tested in other repos and they only ever tested 1 of either true or false, which is why I did the same.

Comment (paulanderson1234) [reply test/unit/uk/gov/hmrc/agentclientmandate/controllers/agent/PaySAQuestionControllerSpec.scala:83]: No problem - we'll just go with some data in that case

File: hs_err_pid29104.log (hunk 1 above)
Comment (paulanderson1234) [reply hs_err_pid29104.log]: File removed from PR

**Pull request merged**